src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Hover/HoverInfoServiceTest.cs

```diff
@@ -477,6 +477,58 @@ public void GetHoverInfo_TagHelper_PlainTextElement_EndTag()
         Assert.Equal(expectedRange, hover.Range);
     }
 
+    [Fact]
+    public void GetHoverInfo_TagHelper_TextComponent()
+    {
+        // Arrange
+        var txt = """
+                <$$Text></Text>
+                """;
+        TestFileMarkupParser.GetPosition(txt, out txt, out var cursorPosition);
+
+        var codeDocument = CreateCodeDocument(txt, isRazorFile: true, DefaultTagHelpers);
+
+        var service = GetHoverInfoService();
+        var location = new SourceLocation(cursorPosition, -1, -1);
+
+        // Act
+        var hover = service.GetHoverInfo("file.razor", codeDocument, location, CreatePlainTextCapabilities());
+
+        // Assert
+        Assert.Contains("Text", ((MarkupContent)hover.Contents).Value, StringComparison.Ordinal);
+        Assert.Equal(MarkupKind.PlainText, ((MarkupContent)hover.Contents).Kind);
+        var expectedRange = new Range
+        {
+            Start = new Position(0, 1),
+            End = new Position(0, 5),
+        };
+        Assert.Equal(expectedRange, hover.Range);
+    }
+
+    [Fact]
+    public void GetHoverInfo_TagHelper_TextComponent_NestedInCSharp()
+    {
+        // Arrange
+        var txt = """
+                @if (true)
+                {
+                    <$$Text></Text>
```

Comment:
Consider testing `<text><$$Text>...` and `<div><$$Text>...` inside C# code.

Reply:
Added, but really its just validating your syntax tree at this point, so as long as you folks don't stuff that up, I'm pretty confident we'll be fine :)